internal/datanode/timetick_sender.go

```diff
@@ -156,27 +142,19 @@ func (m *timeTickSender) mergeDatanodeTtMsg() ([]*msgpb.DataNodeTtMsg, map[strin
 func (m *timeTickSender) cleanStatesCache(sendedLastTss map[string]uint64) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	sizeBeforeClean := len(m.channelStatesCaches)
+	sizeBeforeClean := len(m.channelStatsCache)
 	log := log.With(zap.Any("sendedLastTss", sendedLastTss), zap.Int("sizeBeforeClean", sizeBeforeClean))
 	for channelName, sendedLastTs := range sendedLastTss {
-		channelCache, ok := m.channelStatesCaches[channelName]
-		if ok {
-			for ts := range channelCache.data {
-				if ts <= sendedLastTs {
-					delete(channelCache.data, ts)
-				}
-			}
-			m.channelStatesCaches[channelName] = channelCache
-		}
-		if len(channelCache.data) == 0 {
-			delete(m.channelStatesCaches, channelName)
+		segStats, ok := m.channelStatsCache[channelName]
+		if ok && segStats.ts <= sendedLastTs {
```

Comment:
does that correct?
what if update happens during report timetick? 

we can only clean segmentInfos before report ts?

Reply:
if update happens during report timetick, `segStats.ts > sendedLastTs`, stats cache will not be clean.